pkg/scheduler/cache/event_handlers.go

```diff
@@ -411,24 +411,18 @@ func (sc *SchedulerCache) AddOrUpdateCSINode(obj interface{}) {
 		return
 	}
 
-	var csiNodeStatus *schedulingapi.CSINodeStatusInfo
-	var found bool
+	csiNodeStatus := &schedulingapi.CSINodeStatusInfo{
+		CSINodeName:  csiNode.Name,
+		DriverStatus: make(map[string]bool),
+	}
 	sc.Mutex.Lock()
 	defer sc.Mutex.Unlock()
-	// update nodeVolumeCount
-
-	if csiNodeStatus, found = sc.CSINodesStatus[csiNode.Name]; !found {
-		csiNodeStatus = &schedulingapi.CSINodeStatusInfo{
-			CSINodeName:  csiNode.Name,
-			DriverStatus: make(map[string]bool),
-		}
-		sc.CSINodesStatus[csiNode.Name] = csiNodeStatus
-	}
 
 	for i := range csiNode.Spec.Drivers {
 		d := csiNode.Spec.Drivers[i]
 		csiNodeStatus.DriverStatus[d.Name] = d.Allocatable != nil && d.Allocatable.Count != nil
 	}
+	sc.CSINodesStatus[csiNode.Name] = csiNodeStatus
```

Comment:
if csiNodeStatus exists in sc.CSINodesStatus[csiNode.Name], we still create a new csiNodeStatus?

Reply:
Yes, the old status should be override completely because we don't know which driver pod is dead or recreated, otherwise there may left some non existent driver status as true.

Reply:
Could we just remove the nonexistent drivers in existing `csiNodeStatus`, like:

```go
	for _, driver := range csiNode.Spec.Drivers {
		if _, exist := csiNodeStatus.DriverStatus[driver.Name]; !exist {
			delete(csiNodeStatus.DriverStatus, driver.Name)
			continue
		}
		csiNodeStatus.DriverStatus[driver.Name] = driver.Allocatable != nil && driver.Allocatable.Count != nil
	}
```